nixos/platform/nix.nix

```
@@ -0,0 +1,36 @@
+{ config, lib, pkgs, ... }:
+
+let
+  location = lib.attrByPath [ "parameters" "location" ] "" config.flyingcircus.enc;
+  production = lib.attrByPath [ "parameters" "production" ] "" config.flyingcircus.enc;
+
+  nixPackage = if config.flyingcircus.nix.useUnstableNix
+    then pkgs.nixVersions.nix_2_25
+    else pkgs.nixVersions.nix_2_18;
+in {
+  options.flyingcircus = {
+    nix.useUnstableNix = lib.mkOption {
+      default = (location == "dev") || (location == "whq") || (production  == false);
+      defaultText = lib.literalExpression ''(location == "dev") || (location == "whq") || (production  == false)'';
```

@osnyx you mentioned that you'd prefer 2.18 on staging first. This commit is the state I'd like to see in production at some point.

Would it make sense to you to push another commit on top that sets the default to `false` and adjusts the test accordingly? We could just revert the commit then when this can go to production.

---

Another suggestion would be to also split whq and dev machines depending on whether those are production or not. At least on the VM side we have both prod and non-prod machines in dev and WHQ as well.
```suggestion
      defaultText = lib.literalExpression ''production  == false'';
```

This obviously reduces the number of hosts exposed to Nix 2.25 again, especially for hardware (almost all of our infra is marked as production). But combining this with *push another commit on top*, we could go with my suggestion for this release cycle, and possibly move to the one you suggested later on.

---

As we have 2 releasetest VMs per branch, I also suggest configuring one of them to use the stable nix for the time being.